simpl/server.py

```diff
@@ -0,0 +1,158 @@
+
+
+class EventletLogFilter(object):  # pylint: disable=R0903
+
+    """Receives eventlet log.write() calls and routes them.
+
+    Thread and WSGI messages only get written in debug mode.
+    API calls get written as INFO and to an access_log if provided.
+
+    An instance of this class can be passed in to a bottle.run command using
+    the `log` keyword.
+    """
+
+    def __init__(self, conf, log, access_log=None):
+        """Initialize with config and optional access_log."""
+        self.config = conf
+        self.log = log
+        self.access_log = access_log
+
+    def write(self, text):
+        """Write to appropriate target."""
+        if text:
+            if text[0] in '(w':
+                # write thread and wsgi messages to debug only
+                self.log.debug(text[:-1])
+                return
+            if self.access_log:
+                self.access_log.write(text)
+            self.log.info(text[:-1])
+
+
+class XEventletServer(bottle.ServerAdapter):
+
+    r"""Eventlet Bottle Server Adapter with extensions.
+
+    Supports SSL. Accepts additional tuning parameters:
+
+    * `backlog` adjust the eventlet backlog parameter which is the maximum
+      number of queued connections. Should be at least 1; the maximum
+      value is system-dependent.
+    * `family`: (default is 2) socket family, optional. See socket
+      documentation for available families.
+    * `**kwargs`: directly map to python's ssl.wrap_socket arguments from
+      https://docs.python.org/2/library/ssl.html#ssl.wrap_socket and
+      wsgi.server arguments from
+      http://eventlet.net/doc/modules/wsgi.html#wsgi-wsgi-server
+
+    To create a self-signed key and start the eventlet server using SSL::
+
+      openssl genrsa -des3 -out server.orig.key 2048
+      openssl rsa -in server.orig.key -out test.key
+      openssl req -new -key test.key -out server.csr
+      openssl x509 -req -days 365 -in server.csr -signkey test.key -out \
+      test.crt
+
+      bottle.run(server='eventlet', keyfile='test.key', certfile='test.crt')
+    """
+
+    def get_socket(self):
+        """Create listener socket based on bottle server parameters."""
+        import eventlet
+
+        # Separate out socket.listen arguments
+        socket_args = {}
+        for arg in ('backlog', 'family'):
+            try:
+                socket_args[arg] = self.options.pop(arg)
+            except KeyError:
+                pass
+        # Separate out wrap_ssl arguments
+        ssl_args = {}
+        for arg in ('keyfile', 'certfile', 'server_side', 'cert_reqs',
+                    'ssl_version', 'ca_certs', 'do_handshake_on_connect',
+                    'suppress_ragged_eofs', 'ciphers'):
+            try:
+                ssl_args[arg] = self.options.pop(arg)
+            except KeyError:
+                pass
+        address = (self.host, self.port)
+        try:
+            sock = eventlet.listen(address, **socket_args)
+        except TypeError:
```

Comment:
This is ported from https://github.com/bottlepy/bottle/blob/master/bottle.py#L2997. Based on the comment I would say some version of eventlet.